modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -494,24 +492,15 @@ private PartialNode startPartialNode(
                 fut -> new TestConfigurationCatchUpListener(cfgStorage, fut, revisionCallback0)
         );
 
-        CompletableFuture<?> notificationFuture = CompletableFuture.allOf(
+        CompletableFuture<?> startFuture = CompletableFuture.allOf(
                 nodeCfgMgr.configurationRegistry().notifyCurrentConfigurationListeners(),
-                clusterCfgMgr.configurationRegistry().notifyCurrentConfigurationListeners()
+                clusterCfgMgr.configurationRegistry().notifyCurrentConfigurationListeners(),
+                // Deploy all registered watches because all components are ready and have registered their listeners.
+                metaStorageMgr.deployWatches(),
+                configurationCatchUpFuture
         );
 
-        CompletableFuture<?> startFuture = notificationFuture
-                .thenCompose(v -> {
-                    // Deploy all registered watches because all components are ready and have registered their listeners.
-                    try {
-                        metaStorageMgr.deployWatches();
-                    } catch (NodeStoppingException e) {
-                        throw new CompletionException(e);
-                    }
-
-                    return configurationCatchUpFuture;
-                });
-
-        assertThat(startFuture, willCompleteSuccessfully());
+        assertThat("Partial node was not started", startFuture, willCompleteSuccessfully());
```

Comment:
Why you've changed the order of the processes?

Reply:
I just add a message here, because the assertion relates to deploy watches future.

Reply:
Really? Seems that originally there was notificationFuture(notifyCurrentConfigurationListeners, notifyCurrentConfigurationListeners).thenCompose(deployWatches) and now it's startFuture(notifyCurrentConfigurationListeners, notifyCurrentConfigurationListeners, deployWatches) in other words originally deployWatches was called after notificationFutures and now it's called concurrently. Am I mistaken?